src/js/index.tsx

```diff
@@ -54,6 +55,11 @@ const App = () => (
               />
               <Route path={routePatterns.auth_error()} component={AuthError} />
               <PrivateRoute component={FourOhFour} />
+              <PrivateRoute
+                exact
+                path={routePatterns.product_detail()}
+                component={ProductDetail}
+              />
```

Comment:
@jgerigmeyer am I missing something here?  Some of these patterns aren't in MD for me to follow, but, it's still showing the 404 page when routing to the newly created `ProductDetail` route.

Reply:
@duggiemitchell This route needs to come _before_ the 404 catch-all route (probably should come right after `product_list`). Because the 404 has no `path` defined, it matches any and all urls that get to that point.

Reply:
OK thank you, that seemed to do it; I wasn't sure whether the order mattered.